assets/js/hooks/useDebouncedState.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * useDebouncedState hook
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} value The value to be debounced.
+ * @param {number} delay Number of milliseconds to debounce
+ * @return {string} The update value after the delay
+ */
+export function useDebouncedState( value, delay ) {
```

Comment:
While this is a fairly straightforward implementation, it would be good to add a test for this.

It should be rather straightforward to test as well using `renderHook` and Jest's fake timers API.

We have one test for a custom hook so far (see `useExistingTagEffect`) but that one is much more complicated than this.

Reply:
Added.